### ISSUE TITLE
CA1823 false positive

### ISSUE THREAD

**Issue by kentcb** [dotnet/roslyn-analyzers#1321]
#### Analyzer package

Microsoft.CodeAnalysis.FxCopAnalyzers

#### Analyzer

CA1823

#### Repro steps

```C#
private sealed class SomeClass
{
    private const string someKey = "foo";

    public bool SomeProperty
    {
        get => this.settings.GetValueOrDefault(someKey, true);
        set => this.settings.AddOrUpdateValue(showKeyDatesKey, value);
    }
}
```

#### Expected behavior

No violation reported.

#### Actual behavior

Violation reported:
```
warning CA1823: Unused field 'someKey'.
```

**Comment by zvirja** [comment]
@jinujoseph @mavasani Could you please prioritize this one? It prevents us from usage of new C# features making the code base polluted by suppression.

**Comment by mavasani** [comment]
Sure will investigate today

**Comment by mavasani** [comment]
Verified this works fine with latest FxCopAnalyzers preview package https://dotnet.myget.org/feed/roslyn-analyzers/package/nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/2.6.0-beta1-62231-02 on latest VS internal drop. @jinujoseph Do we know what is that version for the latest VS2017.5 preview?

**Comment by mavasani** [comment]
Trying out this package with latest preview: https://www.visualstudio.com/en-us/news/releasenotes/vs2017-preview-relnotes

**Comment by mavasani** [comment]
Confirmed issue no longer repros with https://dotnet.myget.org/feed/roslyn-analyzers/package/nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/2.6.0-beta1-62231-02 on Visual Studio version 15.5 Preview 3 https://www.visualstudio.com/en-us/news/releasenotes/vs2017-preview-relnotes

**Comment by zvirja** [comment]
@mavasani Thank you a lot for the verification and confirmation! :star2:

Meaning, that we should simply wait for the NuGet release, which is expected in the nearby month and there issue will be no longer present..

Likely, we can close the issue (it's still better to rename it) as no actions are required.

Thank you again for your help!
